### PR TITLE
fix(cachekey): frame exclusion regexes

### PR DESCRIPTION
**Pull request by sywhang**
Length-frame sorted exclusion regexes before hashing so structurally distinct lists cannot produce the same cache key while preserving set-order independence.

Add a Rapid property test that constructs concatenation-boundary aliases, permutes both lists, and verifies distinct keys without mutating inputs.

### CODE REVIEW

File: core/cachekey/cachekey.go [hunk 1]
```diff
@@ -75,7 +78,13 @@ func hashExcludeFilesRegex(excludeFilesRegex []string) string {
 	sort.Strings(sorted)
 	h := md5.New()
 	for _, r := range sorted {
-		h.Write([]byte(r))
+		writeFramedString(h, r)
 	}
 	return fmt.Sprintf("%x", h.Sum(nil))
 }
+
+func writeFramedString(h hash.Hash, value string) {
```
Comment (xytan0056) [inline core/cachekey/cachekey.go:86]: smartt

**Review by xytan0056 — APPROVED** [review]
